rpc/Transceiver.java

```diff
@@ -116,22 +117,24 @@ public Response sendAndReceive(Transaction.Req request) throws InterruptedExcept
     }
 
     public void sendAndReceiveAsync(Transaction.Req request, ResponseReceiver receiver) {
-        executorService.execute(() -> {
-            lock.lock();
-            try {
-                send(request);
-                while (true) {
-                    Response res = receive();
-                    if (!receiver.onResponse(res)) {
-                        break;
+        try (ThreadTrace parentTrace = GrablTracingThreadStatic.traceOnThread("asyncQueue")) {
+            executorService.execute(() -> {
+                lock.lock();
+                try (ThreadTrace ignored = parentTrace.traceOnThread("asyncExecute")) {
```

Comment:
why is this called ignored?

Reply:
but unintuitive

Reply:
`ignored` is a "reserved" variable name that disables warnings automatically. Anonymous resources are not available until a later version of Java.

Reply:
lol never knew this, okay